c++/include/orc/Writer.hh

```diff
@@ -0,0 +1,294 @@
+
+  public:
+    WriterOptions();
+    WriterOptions(const WriterOptions&);
+    WriterOptions(WriterOptions&);
+    WriterOptions& operator=(const WriterOptions&);
+    virtual ~WriterOptions();
+
+    /**
+     * Set the strip size.
+     */
+    WriterOptions& setStripeSize(uint64_t size);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default value.
+     */
+    uint64_t getStripeSize() const;
+
+    /**
+     * Set the block size.
+     */
+    WriterOptions& setBlockSize(uint64_t size);
+
+    /**
+     * Get the block size.
+     * @return if not set, return default value.
+     */
+    uint64_t getBlockSize() const;
+
+    /**
+     * Set row index stride.
+     */
+    WriterOptions& setRowIndexStride(uint64_t stride);
+
+    /**
+     * Get the index stride size.
+     * @return if not set, return default value.
+     */
+    uint64_t getRowIndexStride() const;
+
+    /**
+     * Set the buffer size.
+     */
+    WriterOptions& setBufferSize(uint64_t size);
+
+    /**
+     * Get the buffer size.
+     * @return if not set, return default value.
+     */
+    uint64_t getBufferSize() const;
+
+    /**
+     * Set the dictionary key size threshold.
+     * 0 to disable dictionary encoding.
+     * 1 to always enable dictionary encoding.
+     */
+    WriterOptions& setDictionaryKeySizeThreshold(double val);
+
+    /**
+     * Get the dictionary key size threshold.
+     */
+    double getDictionaryKeySizeThreshold() const;
+
+    /**
+     * Set whether or not to have block padding.
+     */
+    WriterOptions& setBlockPadding(bool padding);
+
+    /**
+     * Get whether or not to have block padding.
+     * @return if not set, return default value which is false.
+     */
+    bool getBlockPadding() const;
+
+    /**
+     * Set Run length encoding version
+     */
+    WriterOptions& setRleVersion(RleVersion version);
+
+    /**
+     * Get Run Length Encoding version
+     */
+    RleVersion getRleVersion() const;
+
+    /**
+     * Set compression kind.
+     */
+    WriterOptions& setCompression(CompressionKind comp);
+
+    /**
+     * Get the compression kind.
+     * @return if not set, return default value which is ZLIB.
+     */
+    CompressionKind getCompression() const;
+
+    /**
+     * Set the encoding strategy.
+     */
+    WriterOptions& setEncodingStrategy(EncodingStrategy strategy);
+
+    /**
+     * Get the encoding strategy.
+     * @return if not set, return default value which is SPEED.
+     */
+    EncodingStrategy getEncodingStrategy() const;
+
+    /**
+     * Set the compression strategy.
+     */
+    WriterOptions& setCompressionStrategy(CompressionStrategy strategy);
+
+    /**
+     * Get the compression strategy.
+     * @return if not set, return default value which is speed.
+     */
+    CompressionStrategy getCompressionStrategy() const;
+
+    /**
+     * Set the writer version.
+     */
+    WriterOptions& setWriterVersion(WriterVersion version);
```

Comment:
The user doesn't get to choose the WriterVersion. It is based on the version of the software that writes the file.